lib/acts_as_list/active_record/acts/shuffle_positions_on_intermediate_items__method_definer.rb

```diff
@@ -0,0 +1,57 @@
+module ActiveRecord::Acts::List::ShufflePositionsOnintermediateItemsDefiner #:nodoc:
+  def self.call(caller_class, sequential_updates)
+    caller_class.class_eval do
+      # Reorders intermediate items to support moving an item from old_position to new_position.
+      # unique constraint prevents regular increment_all and forces to do increments one by one
+      # http://stackoverflow.com/questions/7703196/sqlite-increment-unique-integer-field
+      # both SQLite and PostgreSQL (and most probably MySQL too) has same issue
+      # that's why *sequential_updates* flag is here
+      define_method :shuffle_positions_on_intermediate_items do |old_position, new_position, avoid_id = nil|
```

Comment:
Does defining the method in this way make it a public method? If so, it needs to be private since it currently is.

Also I believe you don't need to use `define_method` since the method name isn't dynamic. You can just use `def shuffle_positions_on_intermediate_items(old_position, new_position, avoid_id = nil)`. You may also just be able to call `private` before the method definition to make it a private method though you'll need to test that. I'm just guessing.